packages/components/src/navigation/components-composition-navigation.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState, useRef } from '@wordpress/element';
+import { Icon, chevronLeft, chevronRight } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Animate from '../animate';
+import {
+	BackButtonUI,
+	MenuItemTitleUI,
+	MenuItemUI,
+	MenuTitleUI,
+	MenuUI,
+	Root,
+} from './styles/navigation-styles';
+import Button from '../button';
+
+export default function ComponentsCompositionNavigation( {
+	initialActiveLevel,
+	children,
+} ) {
+	const [ activeLevel, setActiveLevel ] = useState( initialActiveLevel );
+	const [ slideOrigin, setSlideOrigin ] = useState( 'left' );
+
+	const isMounted = useRef( false );
+	useEffect( () => {
+		if ( ! isMounted.current ) {
+			isMounted.current = true;
+		}
+	}, [] );
+
+	const navigateTo = ( level ) => {
```

Comment:
Nitpicky, but I'd prefer `setLevel` here as `navigateTo` could be ambiguous with navigating to a page/item.